bundle/Form/InformationCollectionUpdateType.php

```diff
@@ -0,0 +1,128 @@
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+use function array_keys;
+use function in_array;
+
+class InformationCollectionUpdateType extends AbstractContentType
+{
+    protected FieldTypeHandlerRegistry $fieldTypeHandlerRegistry;
+
+    protected DataMapperInterface $dataMapper;
+
+    protected ConfigResolverInterface $configResolver;
+
+    public function __construct(
+        FieldTypeHandlerRegistry $fieldTypeHandlerRegistry,
+        DataMapperInterface $dataMapper,
+        ConfigResolverInterface $configResolver
+    ) {
+        parent::__construct($fieldTypeHandlerRegistry, $dataMapper);
+        $this->configResolver = $configResolver;
+    }
+
+    public function getName(): string
+    {
+        return $this->getBlockPrefix();
+    }
+
+    /**
+     * Returns the prefix of the template block name for this type.
+     */
+    public function getBlockPrefix(): string
+    {
+        return 'ezforms_information_collection_update';
+    }
+
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setRequired('collection');
+        $resolver->setAllowedTypes('collection', Collection::class);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        /** @var DataWrapper $dataWrapper */
+        $dataWrapper = $options['data'];
+        $collection = $options['collection'];
+
+        if (!$dataWrapper instanceof DataWrapper) {
+            throw new RuntimeException(
+                'Data must be an instance of Netgen\\EzFormsBundle\\Form\\DataWrapper'
+            );
+        }
+
+        /** @var InformationCollectionStruct $payload */
+        $payload = $dataWrapper->payload;
+
+        if (!$payload instanceof InformationCollectionStruct) {
+            throw new RuntimeException(
+                'Data payload must be an instance of Netgen\\Bundle\\EzFormsBundle\\Form\\Payload\\InformationCollectionStruct'
+            );
+        }
+
+        /** @var ContentType $contentType */
+        $contentType = $dataWrapper->definition;
+
+        if (!$contentType instanceof ContentType) {
+            throw new RuntimeException(
+                'Data definition must be an instance of eZ\\Publish\\API\\Repository\\Values\\ContentType\\ContentType'
+            );
+        }
+
+        $builder->setDataMapper($this->dataMapper);
+
+        foreach ($contentType->getFieldDefinitions() as $fieldDefinition) {
+            if ($fieldDefinition->fieldTypeIdentifier === 'ezuser') {
+                continue;
+            }
+
+            if (!$fieldDefinition->isInfoCollector) {
+                continue;
+            }
+
+            $handler = $this->fieldTypeHandlerRegistry->get($fieldDefinition->fieldTypeIdentifier);
+
+            $handler->buildFieldUpdateForm($builder, $fieldDefinition, $dataWrapper->target->content, $this->getLanguageCode($contentType));
```

Comment:
The line is too long and unreadable, put the params on separate lines.

Reply:
Also, what is `$dataWrapper->target` in this case? Shouldn't that be the `Content` instance?

Reply:
$dataWrapper->target returns a proxy object, we agreed to keep the logic